src/contract/HandlerBasedContract.ts

```diff
@@ -394,6 +387,14 @@ export class HandlerBasedContract<State> implements Contract<State> {
       this.warp.environment === 'testnet',
       reward
     );
+
+    if (!this._evaluationOptions.internalWrites && strict) {
```

Comment:
Move it here

Reply:
hmm, at first glance - I guess I don't understand what issue this move is fixing?

Reply:
Before fix we were calling:
`const handlerResult = await this.callContract(input, undefined, undefined, tags, transfer, strict, vrf);`
notice second param (caller) is undefined, so then here:
https://github.com/warp-contracts/warp/blob/main/src/contract/HandlerBasedContract.ts#L581 we extracting owner by creating signature.

This move is fixing that we are calling now:
`const handlerResult = await this.callContract(input, interactionTx.owner, undefined, tags, transfer, strict, vrf);`
notice second param (caller) is extract from transaction so we don't enter this branch:
https://github.com/warp-contracts/warp/blob/main/src/contract/HandlerBasedContract.ts#L581

Reply:
oh, now I get it! thanks :-)

Reply:
Hmm.. if the strict logic is moved after interaction creation can't we remove these lines:
```
      const handlerResult = await this.callContract(input, undefined, undefined, tags, transfer, strict, vrf);

      if (strict && handlerResult.type !== 'ok') {
        throw Error(`Cannot create interaction: ${handlerResult.errorMessage}`);
      }
```
in `internalWrites` condition? https://github.com/warp-contracts/warp/blob/main/src/contract/HandlerBasedContract.ts#L344

In this case `callContract` would be always called with owner defined and I guess we could remove then this dummy interaction. This would however required evaluating inner writes before strict mode can throw an error. I'm not an expert in internal writes feature, maybe it can have some performance issues.